docs/routeUrl.md

```diff
@@ -1,66 +1,90 @@
+    ViewModel: {
+      recipe: {
+        default() {
+          return {name: "apple pie"};
+        }
+      }
+    }
+  });
+  </script>
+  ```
+  @codepen
 
-```html
-{{ makeLink( "details", routeUrl(page='todos', true) ) }}
-```
+  This produces (with no pretty routing rules):
 
-@signature `{{ routeUrl([merge,] hashes...) }}`
+  ```html
+  <a href='#!&page=recipe&id=5'>{{recipe.name}}</a>
+  ```
 
-Passes the hashes to `route.url` and returns the result.
+  @param {can-stache/expressions/hash} [hashes...] A hash expression like `page='edit' recipeId=id`.
 
-```html
-<a href="{{ routeUrl(page='todos' id=todo.id) }}">details</a>
-```
+  @param {Boolean} [merge] Pass `true` to create a url that merges `hashes` into the
+  current [can-route] properties.  
 
-@param {Boolean} [merge] Pass `true` to create a url that merges `hashes` into the
-current [can-route] properties.  
+  @return {String} Returns the result of calling `route.url`.
 
-@param {can-stache/expressions/hash} [hashes...] A hash expression like `page='edit' recipeId=id`.
+@signature `{{ routeUrl([merge,] hashes...) }}`
 
-@return {String} Returns the result of calling `route.url`.
+  Passes the hashes to `route.url` and returns the result.
+  Using call expressions/parenthesis lets you pass the `merge` option to `route`.  This
+  lets you write a url that only changes specified properties. The example below clicking __recipes__ will add a `page` hash, and clicking on the recipe name will add the `id`:
+
+  ```html
+  <mock-url></mock-url>
+  <cooking-example></cooking-example>
+  <script type="module">
+  import {Component} from "can";
+  import "//unpkg.com/mock-url@^5";
+
+  Component.extend({
+    tag: "cooking-example",
+    view: `<a href='{{ routeUrl(true, page="recipes") }}'>recipes</a>
+      <a href='{{ routeUrl(true, id=5) }}'>{{recipe.name}}</a>`,
+    ViewModel: {
+      recipe: {
+        default() {
+          return {name: "apple pie"};
+        }
+      }
+    }
+  });
+  </script>
+  ```
+  @codepen
+
+  @param {Boolean} [merge] Pass `true` to create a url that merges `hashes` into the
+  current [can-route] properties.  
+
+  @param {can-stache/expressions/hash} [hashes...] A hash expression like `page='edit' recipeId=id`.
+
+  @return {String} Returns the result of calling `route.url`.
 
 @body
 
 ## Use
 
-Use the `routeUrl` helper like:
-
-```html
-<a href='{{ routeUrl(page="recipe" id=5) }}'>{{recipe.name}}</a>
-```
-
-This produces (with no pretty routing rules):
-
-```html
-<a href='#!&page=recipe&id=5'>{{recipe.name}}</a>
-```
-
-Using call expressions/parenthesis lets you pass the `merge` option to `route`.  This
-lets you write a url that only changes specified properties:
-
-```html
-<a href='{{ routeUrl(id=5, true) }}'>{{recipe.name}}</a>
-```
-
-
-
-
 The following demo uses `routeUrl` and [can-stache-route-helpers.routeCurrent] to
 create links that update [can-route]’s `page` attribute:
 
```

Comment:
make the demo codepen-able

Reply:
Based on your previous suggestion I moved the example into the docs and made it codepen-able.